packages/markdown/remark/src/shiki.ts

```diff
@@ -2,9 +2,10 @@ import { bundledLanguages, createCssVariablesTheme, getHighlighter } from 'shiki
 import { visit } from 'unist-util-visit';
 import type { Properties } from 'hast';
 import type { ShikiConfig } from './types.js';
+import type { HTMLAttributes } from '../../../../packages/astro/types.js';
 
 export interface ShikiHighlighter {
-	highlight(code: string, lang?: string, options?: { inline?: boolean }): string;
+	highlight(code: string, lang?: string, options?: { inline?: boolean }, additionalProps?:HTMLAttributes<'pre'>): string;
```

Comment:
Can we call this `attributes` instead and put this inside the third options parameter? We can also reference it as `Record<string, string>` so we don't reference `HTMLAttributes`. This package should be self-contained and not import things from other packages.